RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderEDProducer.cc

```diff
@@ -0,0 +1,130 @@
+  produces<VectorHitCollectionNew>(offlinestubsTag_ + "Rejected");
+  setupAlgorithm(conf);
+}
+
+VectorHitBuilderEDProducer::~VectorHitBuilderEDProducer() { delete stubsBuilder_; }
+
+void VectorHitBuilderEDProducer::produce(edm::Event& event, const edm::EventSetup& es) {
+  LogDebug("VectorHitBuilderEDProducer") << "VectorHitBuilderEDProducer::produce() begin";
+
+  // get input clusters data
+  edm::Handle<edmNew::DetSetVector<Phase2TrackerCluster1D>> clustersHandle;
+  event.getByToken(clusterProducer_, clustersHandle);
+
+  // create the final output collection
+  std::unique_ptr<edmNew::DetSetVector<Phase2TrackerCluster1D>> outputClustersAccepted(
+      new edmNew::DetSetVector<Phase2TrackerCluster1D>);
+  std::unique_ptr<edmNew::DetSetVector<Phase2TrackerCluster1D>> outputClustersRejected(
+      new edmNew::DetSetVector<Phase2TrackerCluster1D>);
+  std::unique_ptr<VectorHitCollectionNew> outputVHAccepted(new VectorHitCollectionNew());
+  std::unique_ptr<VectorHitCollectionNew> outputVHRejected(new VectorHitCollectionNew());
+
+  if (readytobuild_)
+    stubsBuilder_->initialize(es);
+  else
+    edm::LogError("VectorHitBuilderEDProducer") << "Impossible initialization of builder!!";
+
+  // check on the input clusters
+  stubsBuilder_->printClusters(*clustersHandle);
+
+  // running the stub building algorithm
+  //ERICA::output should be moved in the different algo classes?
+  run(clustersHandle, *outputClustersAccepted, *outputClustersRejected, *outputVHAccepted, *outputVHRejected);
+
+  unsigned int numberOfVectorHits = 0;
+  edmNew::DetSetVector<VectorHit>::const_iterator DSViter;
+  for (DSViter = (*outputVHAccepted).begin(); DSViter != (*outputVHAccepted).end(); DSViter++) {
+    edmNew::DetSet<VectorHit>::const_iterator vh;
+    for (vh = DSViter->begin(); vh != DSViter->end(); ++vh) {
+      numberOfVectorHits++;
+      LogDebug("VectorHitBuilderEDProducer") << "\t vectorhit in output " << *vh << std::endl;
+    }
+  }
+  /*
+  if(numberOfVectorHits > _maxOfflinestubs) {
+    edm::LogError("VectorHitBuilderEDProducer") <<  "Limit on the number of stubs exceeded. An empty output collection will be produced instead.\n";
+    VectorHitCollectionNew empty;
+    empty.swap(outputAcc);
+  }
+*/
+  // write output to file
+  event.put(std::move(outputClustersAccepted), "ClustersAccepted");
+  event.put(std::move(outputClustersRejected), "ClustersRejected");
+  event.put(std::move(outputVHAccepted), offlinestubsTag_ + "Accepted");
+  event.put(std::move(outputVHRejected), offlinestubsTag_ + "Rejected");
+
+  //  LogDebug("VectorHitBuilderEDProducer") << " Executing " << _algoTag << " resulted in " << numberOfVectorHits << ".";
+  LogDebug("VectorHitBuilderEDProducer") << "found\n" << numberOfVectorHits << " .\n";
+}
+
+void VectorHitBuilderEDProducer::setupAlgorithm(edm::ParameterSet const& conf) {
+  if (algoTag_ == "VectorHitBuilderAlgorithm") {
+    stubsBuilder_ = new VectorHitBuilderAlgorithm(conf);
```

Comment:
make `stubsBuilder_` a `unique_ptr` for automatic memory management

Reply:
Done.